### ISSUE TITLE
Sabre declared twice when using davcal

### ISSUE THREAD

**Issue by thegcat** [real-or-random/dokuwiki-plugin-icalevents#21]
We currently have a problem when this plugin and caldav are both installed, as both bring (some) sabre parts. The error message is:

```
PHP message: PHP Fatal error:  Cannot redeclare Sabre\Uri\split() (previously declared in /var/www/dokuwiki/lib/plugins/icalevents/vendor/sabre/uri/lib/functions.php:279) in /var/www/dokuwiki/lib/plugins/davcal/vendor/sabre/uri/lib/functions.php on line 250
```

I am unsure how to resolve this. Would it be possible to only conditionally (i.e. only when not already loaded) load sabre?

**Comment by real-or-random** [comment]
Thanks for the bug report.

> I am unsure how to resolve this. Would it be possible to only conditionally (i.e. only when not already loaded) load sabre?

We could do this but it's not guaranteed to work because icalevents and davcal may depend on different versions of sabre...

I searched the web a little bit and found this here, which describes basically the same issue in the context of WordPress plugins:
https://deliciousbrains.com/php-scoper-namespace-composer-depencies/

This may be the way to go.

@andyboeh, any thoughts on this?

**Comment by andyboeh** [comment]
AFAIK, the two plugins are simply incompatible.
Regarding the scoper: I don't think that helps, since it does not prevent two libraries from trying to live in the same namespace.
Yes, basically either davcal or icalevents could check the presence of either and make use of the other bundled library, but that does not prevent any version conflicts.

One (important) thing to add: Currently, I don't have the resources to do bigger changes in davcal. However, would it be an option for @thegcat to use either of the plugins, since they both provide very similar features with regard to ics subscriptions?

**Comment by thegcat** [comment]
We use icalevents to show a table of events from an ical feed, I haven't tried using davcal for this. Would this be the "client mode" the documentation refers to? It seems this requires an admin to have a prepopulated list of feeds to read and/or an admin to add each feed that needs to be connected to the configuration. It would be less ideal than having users just being able to add an arbitrary URL, but that would probably be doable for us, yes.

**Comment by andyboeh** [comment]
Yes, exactly. You need webdavclient installed, configure the feed URLs there (as admin) and include the feeds as webdav:// in davcal or davcaltable to get a tabular output. 
Of course, this does not prevent any incompatibility but might be an option for you in the meantime.

**Comment by real-or-random** [comment]
> Regarding the scoper: I don't think that helps, since it does not prevent two libraries from trying to live in the same namespace.

I think that's exactly what I does, by prefixing the namespaces:
https://github.com/humbug/php-scoper

**Comment by andyboeh** [comment]
Well, I seem to have misinterpreted the description...you are right, that might solve the issue. As mentioned, I don't really have the time to try it.

**Comment by thegcat** [comment]
> Yes, exactly. You need webdavclient installed, configure the feed URLs there (as admin) and include the feeds as webdav:// in davcal or davcaltable to get a tabular output.
> Of course, this does not prevent any incompatibility but might be an option for you in the meantime.

I'll see about using this, for the moment it seems the problem has vanished? (it also did work a long time without this error, we're not quite sure what prompted it to appear) But yes, this would be workable, so we have a solution for our case.

Thank you a bunch for your help!
